pkg/resources/envoy/configmap.go

```diff
@@ -190,6 +193,89 @@ func generateEnvoyHealthCheckListener(ingressConfig v1beta1.IngressConfig, log l
 	}
 }
 
+func GenerateEnvoyTLSFilterChain(tcpProxy *envoytcpproxy.TcpProxy, brokerFqdn string, log logr.Logger) (*envoylistener.FilterChain, error) {
+
+	tlsContext := &tlsv3.DownstreamTlsContext{
+		CommonTlsContext: &tlsv3.CommonTlsContext{
+			TlsParams: &tlsv3.TlsParameters{
+				TlsMinimumProtocolVersion: tlsv3.TlsParameters_TLSv1_2,
+				TlsMaximumProtocolVersion: tlsv3.TlsParameters_TLSv1_3,
+			},
+			TlsCertificates: []*tlsv3.TlsCertificate{
+				{
+					CertificateChain: &envoycore.DataSource{
+						Specifier: &envoycore.DataSource_Filename{
+							Filename: "/certs/certificate.crt",
+						},
+					},
+					PrivateKey: &envoycore.DataSource{
+						Specifier: &envoycore.DataSource_Filename{
+							Filename: "/certs/private.key",
+						},
+					},
+				},
+			},
+		},
+	}
+	pbTlsContext, err := anypb.New(tlsContext)
+	if err != nil {
+		log.Error(err, "could not marshall envoy tcp_proxy tls config")
+		return nil, err
+	}
+
+	pbstTcpProxy, err := anypb.New(tcpProxy)
+	if err != nil {
+		log.Error(err, "could not marshall envoy tcp_proxy config")
+		return nil, err
+	}
+
+	brokerTcpProxyFilter := &envoylistener.Filter{
+		Name: wellknown.TCPProxy,
+		ConfigType: &envoylistener.Filter_TypedConfig{
+			TypedConfig: pbstTcpProxy,
+		},
+	}
+
+	filterChain := &envoylistener.FilterChain{
+		FilterChainMatch: &envoylistener.FilterChainMatch{
+			ServerNames:       []string{brokerFqdn},
```

Comment:
Why do we need at all the broker address based multiplexing? If the max number of ports supported on LoadBalancer in an environment is too low that can be addressed via the use of multiple ingress controllers.